### PR TITLE
Add policy Restrict Clusterrole for Mutating and Validating Admission Webhooks

### PR DESCRIPTION
**Pull request by nsagark**
…ebhooks policy

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: other/restrict-clusterrole-mutating-validating-admission-webhooks/restrict-clusterrole-mutating-validating-admission-webhooks.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+              value:
+              - admissionregistration.k8s.io
+            - key: "{{ element.resources || '' }}"
+              operator: AnyIn
+              value:
+              - mutatingwebhookconfigurations
+              - validatingwebhookconfigurations
+            - key: "{{ contains(element.verbs[], '*') }}"
+              operator: Equals
+              value: true
```
Comment (chipzoller) [inline other/restrict-clusterrole-mutating-validating-admission-webhooks/restrict-clusterrole-mutating-validating-admission-webhooks.yaml:59]: This is unnecessarily verbose. You can add the `contains()` check into the first `foreach` loop by putting both it and the check for the specific verbs under an `any` block. The first two expressions can remain under the `all`.

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Comment by nsagark** [comment]
Hi @chipzoller I have updated the policy. Please review

**Comment by chipzoller** [comment]
Please resolve all comments when re-requesting review. If changes were pushed to address comments, hit the 'Resolve' button so reviewers know.

File: other/restrict-clusterrole-mutating-validating-admission-webhooks/restrict-clusterrole-mutating-validating-admission-webhooks.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+kind: ClusterPolicy
+metadata:
+  name: restrict-clusterrole-mutating-validating-admission-webhooks
+  annotations:
+    policies.kyverno.io/title: Restrict Clusterrole for Mutating and Validating Admission Webhooks
+    policies.kyverno.io/category: Other
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: ClusterRole
+    policies.kyverno.io/description: >-
+      ClusterRoles that grant write permissions over admission webhook should be minimized to reduce powerful identities in the cluster. This policy checks to ensure write permissions are not provided to admission webhooks.
```
Comment (chipzoller) [inline other/restrict-clusterrole-mutating-validating-admission-webhooks/restrict-clusterrole-mutating-validating-admission-webhooks.yaml:11]: Missing annotations `kyverno.io/kyverno-version` and `kyverno.io/kubernetes-version`. For more details on annotations, see [here](https://github.com/kyverno/policies/wiki/Kyverno-annotations).

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Review by chipzoller — CHANGES_REQUESTED** [review]
This looks good other than updating the digest.

**Comment by nsagark** [comment]
Hi @chipzoller Digest seems to be right. Please take a look.

`$ sha256sum restrict-clusterrole-mutating-validating-admission-webhooks.yaml
f9a1178081876a732a9a43975cdfda007989a8e1d9926281474076f9df48780b *restrict-clusterrole-mutating-validating-admission-webhooks.yaml

Sagar@DESKTOP-VS85098 MINGW64 /c/users/sagar/Downloads/kyverno-prs/policies/other/restrict-clusterrole-mutating-validating-admission-webhooks (restrict-clusterrole-mutating-validating-admission-webhooks)
$ ls -lrht
total 8.0K
-rw-r--r-- 1 Sagar 197121 1.8K Jul 31 09:54 restrict-clusterrole-mutating-validating-admission-webhooks.yaml
-rw-r--r-- 1 Sagar 197121 1.2K Jul 31 09:54 artifacthub-pkg.yml`

**Comment by chipzoller** [comment]
See comment [here](https://github.com/kyverno/policies/pull/1067#issuecomment-2260589118).

**Comment by nsagark** [comment]
Hi @chipzoller I have updated the digest.

**Comment by nsagark** [comment]
Hi @chipzoller Please let me know if anything else is needed here.

**Pull request merged**